pkg/agent/mocks/httpsmock.go

```diff
@@ -0,0 +1,31 @@
+package mocks
```

Comment:
https://github.com/golang/mock/blob/master/gomock/call.go#L111 can this existing mock library be used here?

Reply:
I'll check it out! Thanks!